src/credentials/loginManager.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AwsContext } from '../shared/awsContext'
+import { getAccountId } from '../shared/credentials/accountId'
+import { UserCredentialsUtils } from '../shared/credentials/userCredentialsUtils'
+import { getLogger } from '../shared/logger'
+import { createCredentials } from './credentialsCreator'
+import { CredentialsStore } from './credentialsStore'
+
+export class LoginManager {
+    private readonly credentialsStore: CredentialsStore = new CredentialsStore()
+
+    public constructor(private readonly awsContext: AwsContext) {}
+
+    /**
+     * Establishes a Credentials for the Toolkit to use. Essentially the Toolkit becomes "logged in".
+     * If an error occurs while trying to set up and verify these credentials, the Toolkit is "logged out".
+     */
+    public async login(credentialsId: string): Promise<void> {
+        try {
+            const credentials = await this.credentialsStore.getCredentialsOrCreate(credentialsId, createCredentials)
+            if (!credentials) {
```

Comment:
I don't think this will ever hit in its current state. If a credential isn't found, AWSCredentialProviderChain will throw an error: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CredentialProviderChain.html#resolvePromise-property

Reply:
It shouldn't. This is defensive programming against getCredentialsOrCreate and/or createCredentials ever returning undefined.

Reply:
Perhaps note this in a comment and change the error message? Error message could just be something simple like "No credentials found" that encompasses an undefined return.

Reply:
The code feels self explanatory here, I'm not seeing the value in rehashing what the code says.

I'll update the error though, because "create" does not properly reflect the operation.